mapping/document.go

```diff
@@ -436,6 +440,22 @@ func (dm *DocumentMapping) processProperty(property interface{}, path []string,
 			}
 		}
 	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		if subDocMapping != nil {
+			// index by explicit mapping in case of datetime
+			for _, fieldMapping := range subDocMapping.Fields {
+				if fieldMapping.Type == "datetime" {
+					// This number is a UNIX timestamp
+					// hence must parse as a datetime
+					bounds, isUnixFormat := analysis.UnixTimestampFormats[fieldMapping.DateFormat]
+					if isUnixFormat {
```

Comment:
isUnixFormat -> validFormat maybe? also the common go pattern is to just use "ok" i think, you can use that as well over here.